content/tutorial/tutorial.md

```diff
@@ -321,44 +326,44 @@ class Square extends React.Component {
+React DevTools स्थापित करने के बाद, आप पृष्ठ पर किसी भी तत्व पर राइट-क्लिक कर सकते हैं, डेवलपर टूल्स खोलने के लिए "Inspect" पर क्लिक करें, और React टैब दाईं ओर अंतिम टैब के रूप में दिखाई देगा।
 
-**However, note there are a few extra steps to get it working with CodePen:**
+**हालाँकि, ध्यान दें कि CodePen के साथ काम करने के लिए कुछ अतिरिक्त चरण हैं:**
 
-1. Log in or register and confirm your email (required to prevent spam).
-2. Click the "Fork" button.
-3. Click "Change View" and then choose "Debug mode".
-4. In the new tab that opens, the devtools should now have a React tab.
+1. लॉग इन करें या रजिस्टर करें और अपने ईमेल की पुष्टि करें (स्पैम को रोकने के लिए आवश्यक)।
+2. "Fork" बटन पर क्लिक करें।
+3. "Change View" पर क्लिक करें और फिर "Debug Mode" चुनें।
+4. खुलने वाले नए टैब में, devtools में अब एक React टैब होना चाहिए।
 
-## Completing the Game {#completing-the-game}
+## खेल को पूरा करना {#completing-the-game}
 
-We now have the basic building blocks for our tic-tac-toe game. To have a complete game, we now need to alternate placing "X"s and "O"s on the board, and we need a way to determine a winner.
+अब हमारे पास टिक-टैक-टो गेम के लिए बुनियादी बिल्डिंग ब्लॉक हैं। एक पूरा गेम बनाने के लिए, अब हमें वैकल्पिक रूप से Board पर "X" और "O" को रखने की आवश्यकता है, और हमें विजेता को निर्धारित करने के लिए एक तरीका चाहिए।
 
-### Lifting State Up {#lifting-state-up}
+### लिफ्टिंग state अप {#lifting-state-up}
 
-Currently, each Square component maintains the game's state. To check for a winner, we'll maintain the value of each of the 9 squares in one location.
+वर्तमान में, प्रत्येक Square कौम्पोनॅन्ट खेल की स्थिति को बनाए रखता है। विजेता की जाँच करने के लिए, हम एक स्थान पर 9 वर्गों में से प्रत्येक का मान बनाए रखेंगे।
 
-We may think that Board should just ask each Square for the Square's state. Although this approach is possible in React, we discourage it because the code becomes difficult to understand, susceptible to bugs, and hard to refactor. Instead, the best approach is to store the game's state in the parent Board component instead of in each Square. The Board component can tell each Square what to display by passing a prop, [just like we did when we passed a number to each Square](#passing-data-through-props).
+हम सोच सकते हैं कि Board को Square के state के लिए प्रत्येक Square से पूछना चाहिए। हालांकि यह दृष्टिकोण React में संभव है, हम इसे हतोत्साहित करते हैं क्योंकि कोड को समझना मुश्किल हो जाता है, कीड़े के लिए अतिसंवेदनशील और रिफ्लेक्टर के लिए कठिन। इसके बजाय, सबसे अच्छा तरीका यह है कि खेल के राज्य को प्रत्येक Square के बजाय पैरेंट Board कौम्पोनॅन्ट में स्टोर किया जाए। Board कौम्पोनॅन्ट प्रत्येक Square को props पास करके यह बता सकता है कि किसी प्रस्ताव को पारित करके क्या प्रदर्शित करना है, [ठीक उसी तरह जैसे हमने प्रत्येक Square के लिए एक नंबर पारित किया था](#passing-data-through-props)।
```

Comment:
हम सोच सकते हैं कि Board को Square के state के लिए प्रत्येक Square से पूछना चाहिए। -> grammar?

Reply:
हतोत्साहित -> complex

Reply:
> हम सोच सकते हैं कि Board को Square के state के लिए प्रत्येक Square से पूछना चाहिए। -> grammar?

this sounds fine to me based on the english version of it. Any inputs for this?

Reply:
```
हम सोच सकते हैं कि Board को Square के state के लिए प्रत्येक Square से पूछना चाहिए। हालांकि यह तरीका React में इस्तेमाल करना संभव है, हम इसे न इस्तेमाल करने की सलाह देते हैं क्योंकि कोड को समझना मुश्किल हो जाता है, बग के लिए अतिसंवेदनशील और रीफैक्टरिंग में लिए मुश्किल होती है। इसके बजाय, सबसे अच्छा तरीका यह है कि Game के state को प्रत्येक Square के बजाय पैरेंट Board कौम्पोनॅन्ट में स्टोर किया जाए। Board कौम्पोनॅन्ट प्रत्येक Square props पास करके बता सकता है की उसे क्या दिखाना है, [ठीक उसी तरह जैसे हमने हर Square को एक नंबर दिया](#passing-data-through-props)।
```

How about this?

Reply:
हम सोच सकते हैं कि Board को Square के state के लिए प्रत्येक Square से पूछना चाहिए। हालांकि यह तरीका React में इस्तेमाल करना संभव है, हम इसे न इस्तेमाल करने की सलाह देते हैं क्योंकि कोड को समझना मुश्किल हो जाता है, aur saath hi code bugs ke liye ग्रहणशील ho jata hai aur रीफैक्टरिंग मुश्किल हो jaati है

Reply:
इसके बजाय, सबसे अच्छा तरीका यह है कि Game के state को प्रत्येक Square के बजाय पैरेंट Board कौम्पोनॅन्ट में स्टोर किया जाए।  -> perfect

Reply:
Board कौम्पोनॅन्ट प्रत्येक Square "ko" props पास करके बता सकता है की उसे क्या दिखाना है, [ठीक उसी तरह जैसे हमने हर Square को एक नंबर दिया "था"](#passing-data-through-props)।